google/gax/__init__.py

```diff
@@ -464,10 +468,14 @@ def __iter__(self):
         return self
 
     def next(self):
+        """For Python 2.7 compatibility; see __next__."""
+        return self.__next__()
+
+    def __next__(self):
         """Retrieves the next resource."""
         # pylint: disable=next-method-called
         while not self._current:
-            self._current = self._iterator.next()
+            self._current = next(self._iterator)
```

Comment:
I would expect that it's the common way of doing, since there's really only one right way of implementing a custom iterator in each of Python 2 and 3, but I don't really know enough to be able to call it "canonical".

FWIW, this is also basically the way that is recommended by the writers of the `future` library, which is a well-known library for 2/3 intercompatibility: http://python-future.org/compatible_idioms.html#custom-iterators. (They don't show the aliasing of the old `next()` method, but the presence of that method actually is necessary for Python 2 to recognize that this object is an iterator.)